app/src/main/java/io/pslab/activity/OscilloscopeActivity.java

```diff
@@ -1074,143 +1085,62 @@ public boolean onTouchEvent(MotionEvent event) {
     }
 
     public class CaptureTask extends AsyncTask<String, Void, Void> {
-        ArrayList<Entry> entries;
-        String analogInput;
+        ArrayList<ArrayList<Entry>> entries = new ArrayList<>();
+        Integer noOfChannels;
+        String[] paramsChannels;
 
         @Override
-        protected Void doInBackground(String... params) {
+        protected Void doInBackground(String... strings) {
+            paramsChannels = strings;
```

Comment:
Need to copy the variable?

Reply:
Yes... Because I am using input parameters ```strings``` in onPostExecute() method

Reply:
Ahh I see..